CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.12.0)
 
 project(ni_grpc_device_server
   LANGUAGES C CXX
-  VERSION 1.5.2)
+  VERSION 2.0.0)
```

Comment:
I imagine doing it here in the releases/2.0 branch is correct. Then after we have a 2.0 release published we can update from 1.5.2 in the main branch.

I'd guess we'd also update it to 2.0.0 in main as well post-release but could also see us having some clarification that its a development version instead of a released one...

Reply:
The way we do it for the RTOS repos (e.g., [see here](https://github.com/ni/meta-nilrt/blob/61b06f95243c342711731148054aff34f55706d5/conf/distro/nilrt.conf#L3)) is to set the version in main once we start development on that version. When we cut the release branch, it will already have that version and then we post a PR to bump the version in main afterwards. The main difference there is that the history of when the version bump occurred is preserved in both the release branch, the main branch, and any subsequent release branches. @amstewart - Would you have any advice on this? Relying on you as an "expert" here since you're the RTOS code owner. 

I'm not sure how that would play with the automation or how that compares to other open source projects at NI, but it seems reasonable to expect that if a "release" doesn't exist, the version is under development.

Reply:
Mmm, good point. That seems the better route to take. I'll assume we'll want to go that route and should've updated the version in main to this upcoming release a while ago as we were working towards it. I guess for now I'll update in main to 2.0.0, get that in, and then get it into the releases/2.0 branch from the commit in main instead.

Reply:
My recommendation would be to keep the mainline branch versioned like: `last.released.version+git` and the release branch versioned like `new.release.version`.

eg.
* `main` versioned `1.2.3+git`
* cut a release branch `release-1.3.0` with a commit that changes version to `1.3.0`.
* Put a commit into the `main` ref, reversioning it to `1.3.0+git`.

That way, builders using the mainline branch will always get a binary with a `+git` version suffix, unless they build from the release branch. And because the main branch's *number* components are the last release's version number, you don't have to decide what the next version's number will be until you are ready to cut that release's branch.

Reply:
@amstewart I looked into it a little bit and don't think this VERSION we're using for the project in CMAKE allows for a suffix, just non-negative integers so we couldn't distinguish it with a `+git` suffix in main. See CMAKE project version docs [here](https://cmake.org/cmake/help/latest/command/project.html#options)

Reply:
I could add a DESCRIPTION to the project though with "git" or "Development Version" or something like that to denote it in the main branch if you think that would suffice.

Reply:
This is strange to me. CMake doesn't have an option for version suffixes that I can find (and therefore doesn't fully support semantic versioning). 

This also raises another issue - we don't even tag our binaries with versions when they're built as far as I recall.

Reply:
So I guess we could have something like

![ReleaseExecutable](https://user-images.githubusercontent.com/77176215/211928412-82198ca4-558f-497a-8ca0-5be27f8a4602.png)

for the executable created from a Releases branch and

![MainExecutable](https://user-images.githubusercontent.com/77176215/211928255-9f33a401-2bd3-46ae-bc0f-1f318b62b226.PNG)

Reply:
Not great, but I think that's as close as we'll get with respect to this Executable versioning for Windows.

Reply:
> @amstewart I looked into it a little bit and don't think this VERSION we're using for the project in CMAKE allows for a suffix, just non-negative integers so we couldn't distinguish it with a `+git` suffix in main. See CMAKE project version docs [here](https://cmake.org/cmake/help/latest/command/project.html#options)

Oh. I saw [this implementation](https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L31) in grpc and assumed that it did. But I suppose that is a custom variable they're using, rather than the cmake `VERSION`. Lame.